pkg/keycloak/keycloak.go

```diff
@@ -436,6 +435,38 @@ func (h *Handler) reconcileClients(kc *v1alpha1.Keycloak, kcClient KeycloakInter
 	return nil
 }
 
+func (h *Handler) reconcileUsers(kc *v1alpha1.Keycloak, kcClient KeycloakInterface, objRealm *v1alpha1.KeycloakRealm) error {
+	logrus.Info("reconciling users")
+
+	users, err := kcClient.ListUsers(objRealm.Realm)
+	if err != nil {
+		return err
+	}
+
+	kcUsers := map[string]*v1alpha1.KeycloakUser{}
+	for i := range users {
+		kcUsers[users[i].UserName] = users[i]
+	}
+
+	userPairsList := map[string]*v1alpha1.KeycloakUserPair{}
+	for i := range objRealm.Users {
+		user := objRealm.Users[i]
+		userPairsList[user.UserName] = &v1alpha1.KeycloakUserPair{
+			ObjUser: &user,
+			KcUser:  kcUsers[user.UserName],
+		}
+	}
+
+	for _, userPair := range userPairsList {
+		err := h.reconcileUser(userPair.KcUser, userPair.ObjUser, objRealm.Realm, kcClient)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
Can we change all these range iterators to not use the value but always use the index see https://github.com/aerogear/keycloak-operator/pull/23#discussion_r211545272

Constantly hit issues with this, and you want to pass the correct pointer for userPair.ObjUser down into reconcileUser since we are going to need to update it (add the generated ID)

Reply:
:+1: